src/pkg/rbac/dao/dao.go

```diff
@@ -0,0 +1,152 @@
+package dao
+
+import (
+	"context"
+	"fmt"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/orm"
+	"github.com/goharbor/harbor/src/lib/q"
+	"github.com/goharbor/harbor/src/pkg/rbac/model"
+	"time"
+)
+
+// DAO is the data access object interface for rbac policy
+type DAO interface {
+	// CreatePermission ...
+	CreatePermission(ctx context.Context, rp *model.RolePermission) (int64, error)
+	// DeletePermission ...
+	DeletePermission(ctx context.Context, id int64) error
+	// ListPermission ...
+	ListPermission(ctx context.Context, query *q.Query) ([]*model.RolePermission, error)
+	// DeletePermissionByRole ...
+	DeletePermissionByRole(ctx context.Context, roleType string, roleID int64) error
+
+	// CreateRbacPolicy ...
+	CreateRbacPolicy(ctx context.Context, rp *model.RbacPolicy) (int64, error)
+	// DeleteRbacPolicy ...
+	DeleteRbacPolicy(ctx context.Context, id int64) error
+	// ListRbacPolicy list RbacPolicy according to the query.
+	ListRbacPolicy(ctx context.Context, query *q.Query) ([]*model.RbacPolicy, error)
+
+	// GetPermissionsByRole ...
+	GetPermissionsByRole(ctx context.Context, roleType string, roleID int64) ([]*model.RolePermissions, error)
+}
+
+// New returns an instance of the default DAO
+func New() DAO {
+	return &dao{}
+}
+
+type dao struct{}
+
+func (d *dao) CreatePermission(ctx context.Context, rp *model.RolePermission) (id int64, err error) {
+	ormer, err := orm.FromContext(ctx)
+	if err != nil {
+		return 0, err
+	}
+	rp.CreationTime = time.Now()
+	return ormer.InsertOrUpdate(rp, "role_type, role_id, rbac_policy_id")
+}
+
+func (d *dao) DeletePermission(ctx context.Context, id int64) (err error) {
+	ormer, err := orm.FromContext(ctx)
+	if err != nil {
+		return err
+	}
+	n, err := ormer.Delete(&model.RolePermission{
+		ID: id,
+	})
+	if err != nil {
+		return err
+	}
+	if n == 0 {
+		return errors.NotFoundError(nil).WithMessage("role permission %d not found", id)
+	}
```

Comment:
Not needed in this scenario.

Reply:
should we follow the same pattern? like https://github.com/goharbor/harbor/blob/master/src/pkg/artifact/dao/dao.go#L173

Reply:
discussed offline, keep it.